src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/TasksEmailSender.java

```diff
@@ -0,0 +1,55 @@
+package org.fogbowcloud.saps.engine.core.dispatcher.email;
+
+import com.google.gson.Gson;
+import org.apache.log4j.Logger;
+
+import javax.mail.MessagingException;
+import java.util.*;
+
+public class TasksEmailSender implements Runnable {
+
+    private static final Logger LOGGER = Logger.getLogger(TasksEmailSender.class);
+    private static final String EMAIL_TITLE = "[SAPS] Results of selected tasks";
+
+    private final List<TaskCompleteInfo> tasksEmail;
+    private final Gson gson;
+    private final String userEmail;
+    private final String noReplyEmail;
+    private final String noReplyPass;
+
+    public TasksEmailSender(String noReplyEmail, String noReplyPass, String userEmail,
+                            List<TaskCompleteInfo> tasksEmail) {
+        if (isInvalidFields(noReplyEmail, noReplyPass, userEmail, tasksEmail))
+            throw new IllegalArgumentException("Illegals arguments to use the send email feature");
+
+        this.gson = new Gson();
+        this.noReplyEmail = noReplyEmail;
+        this.noReplyPass = noReplyPass;
+        this.userEmail = userEmail;
+        this.tasksEmail = tasksEmail;
+    }
+
+    private boolean isInvalidFields(String noReplyEmail, String noReplyPass, String userEmail,
+                                    List<TaskCompleteInfo> tasksEmail) {
+        return noReplyEmail.trim().isEmpty() || Objects.isNull(noReplyEmail) ||
```

Comment:
we might get exceptions in this method. For example, we are checking for emptiness before checking for nullity.